src/wp-includes/general-template.php

```diff
@@ -3847,25 +3859,26 @@ function paginate_links( $args = '' ) {
 	if ( $args['prev_next'] && $current && $current < $total ) :
 		$link = str_replace( '%_%', $args['format'], $args['base'] );
 		$link = str_replace( '%#%', $current + 1, $link );
-		if ( $add_args )
+		if ( $add_args ) {
 			$link = add_query_arg( $add_args, $link );
+		}
 		$link .= $args['add_fragment'];
 
 		/** This filter is documented in wp-includes/general-template.php */
-		$page_links[] = '<a class="next page-numbers" href="' . esc_url( apply_filters( 'paginate_links', $link ) ) . '">' . $args['next_text'] . '</a>';
+		$page_links[] = '<a class="' . trim( esc_attr( 'next page-numbers ' . $args['a_classes'] ) ) . '" href="' . esc_url( apply_filters( 'paginate_links', $link ) ) . '">' . $args['next_text'] . '</a>';
 	endif;
 	switch ( $args['type'] ) {
-		case 'array' :
+		case 'array':
 			return $page_links;
 
-		case 'list' :
-			$r .= "<ul class='page-numbers'>\n\t<li>";
-			$r .= join("</li>\n\t<li>", $page_links);
+		case 'list':
+			$r .= "<ul class='page-numbers " . trim( esc_attr( 'page-numbers ' . $args['ul_classes'] ) ) . "'>\n\t<li class='" . esc_attr( $args['li_classes'] ) . "'>";
```

Comment:
This looks broken (`page-numberspage-numbers some-other-class`)

I would expect a test to fail because of this change. We should expand test coverage for the existing code and also make sure the new attributes have tests.